src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder_ListPatterns.cs

```diff
@@ -62,7 +71,13 @@ private Tests MakeTestsAndBindingsForListPattern(BoundDagTemp input, BoundListPa
                         continue;
                     }
 
-                    var indexEvaluation = new BoundDagIndexerEvaluation(subpattern.Syntax, subpattern.InputType, lengthTemp, index++, list.IndexerAccess, list.IndexerSymbol, input);
+                    Debug.Assert(list.IndexerAccess is not null);
+                    Debug.Assert(list.ReceiverPlaceholder is not null);
+                    Debug.Assert(list.ArgumentPlaceholder is not null);
+
+                    var indexEvaluation = new BoundDagIndexerEvaluation(subpattern.Syntax, subpattern.InputType, lengthTemp, index++,
+                        list.IndexerAccess, list.ReceiverPlaceholder, list.ArgumentPlaceholder, input);
```

Comment:
>list.IndexerAccess [](http://example.com/codeflow?start=24&length=18)

Consider asserting what kind of node this can be since the comment for BoundDagIndexerEvaluation narrows the set of nodes that should be expected by comparison to BoundListPattern